unit_test.sh

```diff
@@ -0,0 +1,135 @@
+IMAGE_TAG=$(git rev-parse --short=7 HEAD)
+export IMAGE_TAG
+
+function teardown_docker {
+  docker rm -f "$DB_CONTAINER_ID" || true
+  docker rm -f "$TEST_CONTAINER_ID" || true
+  docker network rm "$NETWORK" || true
+}
+
+trap "teardown_docker" EXIT SIGINT SIGTERM
+
+NETWORK="compliance-test-${IMAGE_TAG}"
+POSTGRES_IMAGE="quay.io/cloudservices/postgresql-rds:cyndi-12-1"
+IMAGE="quay.io/cloudservices/compliance-backend"
+IMAGE_TAG=$(git rev-parse --short=7 HEAD)
+DATABASE_USER="compliance"
+DATABASE_PASSWORD="changeme"
+DATABASE_NAME="compliance-test"
+DATABASE_HOST="compliance-db"
+
+docker network rm "$NETWORK" || echo "network likely doesn't exist"
+docker network create --driver bridge "$NETWORK"
+
+DB_CONTAINER_ID=$(docker run -d \
+  --name "$DATABASE_HOST" \
+  --network "$NETWORK" \
+  -e POSTGRESQL_USER="$DATABASE_USER" \
+  -e POSTGRESQL_PASSWORD="$DATABASE_PASSWORD" \
+  -e POSTGRESQL_DATABASE="$DATABASE_NAME" \
+  "$POSTGRES_IMAGE" || echo "0")
+
+if [[ "$DB_CONTAINER_ID" == "0" ]]; then
+  echo "Failed to start DB container"
+  exit 1
+fi
+
+# Do tests
+TEST_CONTAINER_ID=$(docker run -d \
+  --network "$NETWORK" \
+  -e DATABASE_SERVICE_NAME=postgresql \
+  -e POSTGRESQL_SERVICE_HOST="$DATABASE_HOST" \
+  -e POSTGRESQL_USER="$DATABASE_USER" \
+  -e POSTGRESQL_PASSWORD="$DATABASE_PASSWORD" \
+  -e POSTGRESQL_DATABASE="$DATABASE_NAME" \
+  -e POSTGRESQL_PORT="$DATABASE_PORT" \
+  -e RAILS_ENV=test \
+  "$IMAGE:$IMAGE_TAG" \
+  /bin/bash -c 'sleep infinity' || echo "0")
+
+if [[ "$TEST_CONTAINER_ID" == "0" ]]; then
+  echo "Failed to start test container"
+  exit 1
+fi
+
+WORKSPACE=${WORKSPACE:-./}
+ARTIFACTS_DIR="$WORKSPACE/artifacts"
+mkdir -p "$ARTIFACTS_DIR"
+
+# gem install
+echo '===================================='
+echo '=== Installing Gem Dependencies ===='
+echo '===================================='
+set +e
+docker exec "$TEST_CONTAINER_ID" /bin/bash -c 'bundle install --with test'
+TEST_RESULT=$?
+set -e
+if [[ $TEST_RESULT -ne 0 ]]; then
+  echo '====================================='
+  echo '==== ✖ ERROR: GEM INSTALL FAILED ===='
+  echo '====================================='
+  exit 1
+fi
+
+# setup db
+echo '===================================='
+echo '===     Setting Up Database     ===='
+echo '===================================='
+set +e
+docker exec "$TEST_CONTAINER_ID" /bin/bash -c 'bundle exec rake db:test:prepare'
+TEST_RESULT=$?
+set -e
+if [[ $TEST_RESULT -ne 0 ]]; then
+  echo '====================================='
+  echo '====  ✖ ERROR: DB SETUP FAILED   ===='
+  echo '====================================='
+  exit 1
+fi
+
+# setup cyndi
+echo '===================================='
+echo '===       Setting Up Cyndi      ===='
+echo '===================================='
+set +e
+docker cp "$TEST_CONTAINER_ID":/opt/app-root/src/db/cyndi_setup_test.sql "$WORKSPACE/"
+docker cp "$WORKSPACE/cyndi_setup_test.sql" "$DB_CONTAINER_ID":/var/lib/pgsql/
+rm "$WORKSPACE/cyndi_setup_test.sql"
```

Comment:
I think you should be able to 
```
docker cp $TEST_CONTAINER_ID":/opt/app-root/src/db/cyndi_setup_test.sql "$DB_CONTAINER_ID":/var/lib/pgsql/cyndi_setup_test.sql
```

Reply:
`docker cp` only supports copying into or out of a single container

Reply:
huh ?  Itried this myself with two dummy `busybox` containers. However , locally, and with *podman* . Okay if not possible with Docker then ...

Reply:
Show me your logs:

```
$ podman cp compliance-backend_rails_1:app compliance-backend_inventory-consumer_1:/tmp/
Error: invalid arguments "compliance-backend_rails_1:app", "compliance-backend_inventory-consumer_1:/tmp/": exactly 1 container expected but 2 specified
```

Reply:
```
podman exec -it container-b sh
/ # ls -lrt 
total 0
drwxr-xr-x    2 root     root          4714 Aug 20 16:21 bin
drwxrwxrwt    2 root     root             0 Aug 20 16:21 tmp
drwxr-xr-x    4 root     root            16 Aug 20 16:21 var
drwxr-xr-x    3 root     root             8 Aug 20 16:21 usr
drwxr-xr-x    2 nobody   nobody           0 Aug 20 16:21 home
dr-xr-xr-x   13 nobody   nobody           0 Aug 30 08:11 sys
drwxr-xr-t    3 root     root            80 Sep  2 08:17 run
dr-xr-xr-x  401 nobody   nobody           0 Sep  2 08:17 proc
drwxr-xr-x    3 root     root            56 Sep  2 08:17 etc
drwxr-xr-x    5 root     root           360 Sep  2 08:17 dev
drwx------    2 root     root            24 Sep  2 08:18 root
/ # 
 vmugicag   1362  ~  dev  projects  compliance-backend  
 vmugicag   1362  ~  dev  projects  compliance-backend  podman cp container-a:foo container-b:foo
 vmugicag   1362  ~  dev  projects  compliance-backend  podman exec -it container-b sh
/ # ls -lrt 
total 4
drwxr-xr-x    2 root     root          4714 Aug 20 16:21 bin
drwxrwxrwt    2 root     root             0 Aug 20 16:21 tmp
drwxr-xr-x    4 root     root            16 Aug 20 16:21 var
drwxr-xr-x    3 root     root             8 Aug 20 16:21 usr
drwxr-xr-x    2 nobody   nobody           0 Aug 20 16:21 home
dr-xr-xr-x   13 nobody   nobody           0 Aug 30 08:11 sys
drwxr-xr-t    3 root     root            80 Sep  2 08:17 run
dr-xr-xr-x  400 nobody   nobody           0 Sep  2 08:17 proc
drwxr-xr-x    3 root     root            56 Sep  2 08:17 etc
drwxr-xr-x    5 root     root           360 Sep  2 08:17 dev
-rw-r--r--    1 root     root             6 Sep  2 08:18 foo
drwx------    2 root     root            24 Sep  2 08:18 root
/ # cat foo 
foooo
/ # exit
```

Reply:
That's odd. Which version of podman?

```
$ podman --version
podman version 3.2.3
```

The usage in docker's docs make it seem like it's not supposed to support copying from one container to another. https://docs.docker.com/engine/reference/commandline/cp/

Reply:
```
$ podman --version
podman version 3.3.0
```

Reply:
yeah I also checked the docs, however it doesn't clearly state it is not supported, just shows from container to local, or vice-versa. I tested it myself because it wasn't clear, and when it worked I was a bit shocked. 

However, it's a minor thing, and if it's Docker specific maybe we should leave the middle tempfile